dashboards/model/common.libsonnet

```diff
@@ -0,0 +1,469 @@
+local layout = import 'github.com/grafana/grafonnet-lib/contrib/layout.libsonnet';
+local g = import 'github.com/grafana/grafonnet-lib/grafonnet-7.0/grafana.libsonnet';
+local model = import 'model.libsonnet';
+
+{
+  dashboard(kind, config):
+    g.dashboard.new(title='Explore / %s' % kind, refresh='5m', uid=std.md5(kind))
+    .setTime(from='now-1h', to='now')
+    .addTemplate(
+      g.template.datasource.new(name='datasource', query='prometheus')
+      .setCurrent(text='default', value='default')
+    )
+    .addTemplate(
+      g.template.query.new(
+        datasource='${datasource}',
+        label='cluster',
+        name='cluster',
+        query='label_values(up{%(kubeStateMetricsSelector)s}, cluster)' % config,
+        refresh=1,
+        regex='',
+      )
+    )
+    + layout {
+      // To make building dashboards easier, we use monads(!).
+      //
+      // The idea is that you build a a list of monads to a dashboard by
+      // calling add.
+      //
+      // dashboard('test').chain([
+      //     $.addTemplate('pod'),
+      //     $.addRow('Info'),
+      //     $.addTextRow('Name'),
+      //     $.addLabelPanel('group by (name) (kube_pod_info{pod="$pod"})', 'name'),
+      //     $.newRow,
+      // ])
+      //
+      // This saves you having to nest the return value from each monad.
+      chain(fs):: std.foldl(function(d, f) if f != null then f(d) else d, fs, self),
+    },
+
+  addTemplate(name, config):
+    function(d)
+      d.addTemplate(
+        g.template.query.new(
+          datasource='${datasource}',
+          name=name,
+          query='label_values(up{%(kubeStateMetricsSelector)s, cluster="$cluster"}, %(name)s)' % { kubeStateMetricsSelector: config.kubeStateMetricsSelector, name: name },
+          refresh=1,
+          regex='',
+          hide=2,
```

Comment:
@tomwilkie What is the reasoning behind having these template vars hidden?

I added a `label_values` query before realizing they were hidden, since there does not seem to be a clear "entrypoint" to the explore dashboards, and I had to manually inject a namespace (`var-namespace=kube-system`) on the namespace dashboard to begin exploring.

I would not be opposed to all of the template vars being visible, at least for an 0.1 of this feature, in order to allow people to use it directly, without being linked to one of the explore dashboards from one of the metrics dashboards 🤔

Reply:
> @tomwilkie What is the reasoning behind having these template vars hidden?

I wanted to hide the inner workings to make it seem more "magic", but in retrospect I think we should make them visible.

> here does not seem to be a clear "entrypoint" to the explore dashboards

Yeah, I think we need to build a dashboard that lists all the namespaces across all the datasources as an entry point.